src/modules/iotjs_module_crypto.c

```diff
@@ -0,0 +1,396 @@
+/* Copyright 2018-present Samsung Electronics Co., Ltd. and other contributors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "iotjs_def.h"
+#include "iotjs_handlewrap.h"
+#include "iotjs_module_buffer.h"
+
+
+#if !ENABLE_MODULE_TLS
+
+/*
+ * 32-bit integer manipulation macros (big endian)
+ */
+#ifndef GET_UINT32_BE
+#define GET_UINT32_BE(n, b, i)                                          \
+  {                                                                     \
+    (n) = ((uint32_t)(b)[(i)] << 24) | ((uint32_t)(b)[(i) + 1] << 16) | \
+          ((uint32_t)(b)[(i) + 2] << 8) | ((uint32_t)(b)[(i) + 3]);     \
+  }
+#endif
+
+#ifndef PUT_UINT32_BE
+#define PUT_UINT32_BE(n, b, i)                 \
+  {                                            \
+    (b)[(i)] = (unsigned char)((n) >> 24);     \
+    (b)[(i) + 1] = (unsigned char)((n) >> 16); \
+    (b)[(i) + 2] = (unsigned char)((n) >> 8);  \
+    (b)[(i) + 3] = (unsigned char)((n));       \
+  }
+#endif
+
+static int iotjs_sha1_process(uint32_t state[5], const unsigned char data[64]) {
```

Comment:
Don't forget the license if you take this from somewhere.

Reply:
Right, thank you!